x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/OrderBy.java

```diff
@@ -109,4 +110,9 @@ public void postAnalysisVerification(Failures failures) {
             }
         });
     }
+
+    @Override
+    public void postOptimizationVerification(Failures failures) {
+        failures.add(fail(this, "The query cannot be executed because it would require unbounded sort"));
```

Comment:
I'd rephrase the error message to include an action item for the user: `Unbounded sort not supported yet, please add a limit"